sdks/java/core/src/main/java/org/apache/beam/sdk/io/Sink.java

```diff
@@ -259,6 +259,14 @@ public abstract void finalize(Iterable<WriteT> writerResults, PipelineOptions op
     public abstract WriteT close() throws Exception;
 
     /**
+     * Aborts writing the bundle. Removes resources used for writing the bundle,
+     * and closes this {@link Writer}.
+     *
+     * <p>Called for each failed bundle.
```

Comment:
I do not think this is a correct description of when this function will be called, based on the semantics of TearDown.